src/Markup/Avalonia.Markup.Xaml/MarkupExtensions/ResourceInclude.cs

```diff
@@ -21,8 +22,9 @@ public IResourceDictionary Loaded
         {
             get
             {
-                if (_loaded == null)
+                if (!_isLoading)
```

Comment:
This breaks nullability contract since we claim that we won’t return null. And I think that after your change we will return null.

Reply:
we have to return null, otherwise you just keep recursively looking in the same dictionary.

Reply:
isLoading means you have already started looking in this Resource, try another one.

Reply:
I know why you are doing this but this will cause a warning and potentially crash code that is not expecting a null here. We should update nullability of this property and fix usages.

Reply:
I actually think this is the wrong place for the `isLoading` check. It should be done in `TryGetResource` and return `false` there if `isLoading` is set.

Reply:
This check also needs to be done in `ResourceInclude` i think.

Reply:
@MarchingCube implemented as @grokys suggested and that keeps the nullability contract intact.

Working unit tests also provided courtesy of @grokys